segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/SystemJournal.java

```diff
@@ -552,36 +559,174 @@ private CompletableFuture<Void> writeSnapshotInfo(long snapshotId) {
+                                                    if (existingSnapshot.getEpoch() > epoch) {
+                                                        return CompletableFuture.failedFuture(new StorageNotPrimaryException(String.format("SystemJournal[{}] Unexpected snapshot. Expected = {} actual = {}", info, existingSnapshot)));
+                                                    } else {
+                                                        return chunkStorage.delete(ChunkHandle.writeHandle(snapshotInfoFileName));
+                                                    }
+                                                }, executor);
+                                    } else {
+                                        return CompletableFuture.completedFuture(null);
+                                    }
+                                }, executor)
+                                .thenComposeAsync(v -> chunkStorage.createWithContent(snapshotInfoFileName,
+                                                bytes.getLength(),
+                                                new ByteArrayInputStream(bytes.array(), bytes.arrayOffset(), bytes.getLength())),
+                                        executor)
+                                .thenAcceptAsync(v -> {
+                                    log.debug("SystemJournal[{}] Snapshot info saved to LTS. File {}. info = {}", containerId, snapshotInfoFileName, info);
+                                }, executor)
+                                .thenComposeAsync(v -> readSnapshotInfoFromFile(), executor)
+                                .thenApplyAsync( readBackInfo -> {
+                                    if (readBackInfo.getEpoch() > epoch) {
+                                        return CompletableFuture.failedFuture(
+                                                new StorageNotPrimaryException(String.format("SystemJournal[{}] Unexpected snapshot. Expected = {} actual = {}", info, readBackInfo)));
+                                    }
+                                    if (!info.equals(readBackInfo)) {
+                                        return CompletableFuture.failedFuture(
+                                                new IllegalStateException(String.format("SystemJournal[{}] Unexpected snapshot. Expected = {} actual = {}", info, readBackInfo)));
+                                    }
+                                    return CompletableFuture.completedFuture(null);
+                                }, executor)
+                                .handleAsync((v, e) -> {
+                                    if (null != e) {
+                                        val ex = Exceptions.unwrap(e);
+                                        if (ex instanceof StorageNotPrimaryException) {
+                                            log.warn("SystemJournal[{}] Error while snapshot info to file {}. info = {}", containerId, snapshotInfoFileName, info, e);
```

Comment:
is "saving/writing" after while missing in the log messages in this whole block?